crates/extensions/tedge_http_ext/src/lib.rs

```diff
@@ -13,30 +13,41 @@ use actor::*;
 use tedge_actors::Concurrent;
 use tedge_actors::ServerActorBuilder;
 use tedge_actors::ServerConfig;
+use tedge_config::TEdgeConfig;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct HttpActor {
     config: ServerConfig,
+    tls_client_config: rustls::ClientConfig,
 }
 
 impl HttpActor {
-    pub fn new() -> Self {
-        HttpActor::default()
+    pub fn new(tedge_config: &TEdgeConfig) -> Self {
```

Comment:
> Either we have a struct with all the root cert paths that we generate with a From<&TEdgeConfig> implementation

In the product code, yes, we'll have to do that. I was talking more about avoiding the temp dir creation and all that boilerplate to get a tedge config instantiated in unit tests. The benefit is purely in unit tests.

> Maybe, but purely from the perspective of how many lines of setup code are required, initialising a config struct wouldn't have been quicker.

In most cases, I was just expecting `HttpActorConfig::default()` with an empty cert list. Since its's just one config setting for now, I agree that it's just a matter of 1 vs 3 lines of code. I made that suggestion after seeing the same small boiler plate repeated in 2-3 other tests as well. After dealing with this problem with several actors, as they grew over time, I've involuntarily started introducing actor config structs whenever an actor has had a dependency on tedge config. That's why you'll see that pattern in most of the other actors. But, in this case I leave it up-to-you.

> I think the issue here is more to do with "why do we need to create an entire temporary directory to generate a TEdgeConfig in the tests" than anything else. And most of the tests aren't and shouldn't be like this, unit tests don't need to

Completely agree with you. Using actor specific configs was just our way of getting around that limitation. The ideal solution would have been to fix that, so that we can easily instantiate a tedge config with only the required config settings relevant for the test. Something like `TedgeConfig::default().with_relevant_setting1().with_relevant_setting2()`.